libs/core/thread_support/src/set_thread_name.cpp

```diff
@@ -83,16 +81,40 @@ namespace hpx::util {
         }
     }    // namespace detail
 
+}    // namespace hpx::util
+
+#elif (defined(__linux__))
+
+#include <pthread.h>
+namespace hpx::util { namespace detail {
+    void set_thread_name(char const* thread_name)
+    {
+        pthread_setname_np(pthread_self(), thread_name);
```

Comment:
I think this function is not always available. Should we add a feature test for it? This would also enable the new functionality on other POSIX derivatives (BSD?).

Reply:
> Sure, will do it. Maybe something like a config test sounds good I guess, like the ones in (https://github.com/STEllAR-GROUP/hpx/blob/3d95d662eb6879fccfbc8090377e58674fbc91a7/cmake/tests)

Yes, that's what I meant. FWIW, the feature tests are defined here : https://github.com/STEllAR-GROUP/hpx/blob/master/cmake/HPX_AddConfigTest.cmake, and are invoked here: https://github.com/STEllAR-GROUP/hpx/blob/master/cmake/HPX_PerformCxxFeatureTests.cmake